pkg/kube-metrics/metrics.go

```diff
@@ -0,0 +1,128 @@
+// Copyright 2019 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kubemetrics
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/operator-framework/operator-sdk/pkg/k8sutil"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	kuberuntime "k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/rest"
+	kcollector "k8s.io/kube-state-metrics/pkg/collector"
+	ksmetric "k8s.io/kube-state-metrics/pkg/metric"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+)
+
+var log = logf.Log.WithName("kubemetrics")
+
+// GetGVKsFromAddToScheme takes in the operator specific scheme and
+// filters out all generic apimachinery meta types. It returns the GVK specific to this operator.
+func GetGVKsFromAddToScheme(addToSchemeFunc func(*runtime.Scheme)) ([]schema.GroupVersionKind, error) {
+	s := kuberuntime.NewScheme()
+	addToSchemeFunc(s)
+	operatorKnownTypes := s.AllKnownTypes()
+	operatorGVKs := []schema.GroupVersionKind{}
+	for gvk, _ := range operatorKnownTypes {
+		if !isKubeMetaKind(gvk.Kind) {
+			operatorGVKs = append(operatorGVKs, gvk)
+		}
+	}
+
+	return operatorGVKs, nil
+}
+
+func isKubeMetaKind(kind string) bool {
+	if strings.HasSuffix(kind, "List") ||
+		kind == "GetOptions" ||
+		kind == "DeleteOptions" ||
+		kind == "ExportOptions" ||
+		kind == "APIVersions" ||
+		kind == "APIGroupList" ||
+		kind == "APIResourceList" ||
+		kind == "UpdateOptions" ||
+		kind == "CreateOptions" ||
+		kind == "Status" ||
+		kind == "WatchEvent" ||
+		kind == "ListOptions" ||
+		kind == "APIGroup" {
+		return true
+	}
+
+	return false
+}
+
+// ServeCRMetrics generates CR specific metrics based on the operator GVK.
+// It starts serving collections of those metrics on given host and port.
+func ServeCRMetrics(cfg *rest.Config,
```

Comment:
Should we rename this to `ServeAndGenerateCRMetrics`? Or if anyone has a better naming idea.

Reply:
Switch the order `GenerateAndServeCRMetrics()`?
But even this is clear enough I think.